### PR TITLE
[RFR] Support nested keys and removed keys

### PR DESCRIPTION
**Pull request by KittyGiraudel**

### CODE REVIEW

File: tests/de.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"animals":{"ant":"ant","bat":"bat","cat":"cat"},"colors":{"white":"white","black":"black","yellow":"yellow","pink":"pink"},"winner":"winner"}
```
Comment (schmkr) [inline tests/de.json]: With this formatting, after having run `npm test` the formatting of this file has changed to:

```
{
  "animals": {
    "ant": "ant",
    "bat": "bat",
    "cat": "cat"
  },
  "colors": {
    "white": "white",
    "black": "black",
    "yellow": "yellow",
    "pink": "pink"
  },
  "winner": "winner"
}
```

So maybe commit the file like that?

Comment (KittyGiraudel) [inline tests/de.json]: Actually this file is not being used. I should remove it. :)

**Comment by schmkr** [comment]
So the `object-assign-deep` package used for deep merging does also not handle removed keys from the toDictionary? Is that something you'd consider implementing and if so, how would you do that? Introduce yet another dependency? Wouldn't it be easier to keep a function like the one in #2? (not to support my own PR, but to keep the amount of dependencies to a minimum).

**Comment by KittyGiraudel** [comment]
@schmkr Updated the PR one last time. Tell me what you think. :)

**Comment by schmkr** [comment]
Looks good to me!

**Pull request merged**